shub/config.py

```diff
@@ -370,10 +371,34 @@ def get_image(self, target):
         return image
 
 
-Target = namedtuple('Target', ['project_id', 'endpoint', 'apikey', 'stack',
-                               'image', 'requirements_file', 'version',
-                               'eggs'])
+_Target = namedtuple('Target', ['project_id', 'endpoint', 'apikey', 'stack',
+                                'image', 'requirements_file', 'version',
+                                'eggs'])
+
+
+class APIkey(str):
+
+    def __new__(cls, *args, **kwargs):
+        cls._inst = super(APIkey, cls).__new__(cls, *args, **kwargs)
+        return cls._inst
+
+    def __init__(self, value=None):
+        self.apikey = value
+
+    def __repr__(self):
+        if not self.apikey:
+            return ''
+        visible_chars = APIKEY_SHOW_N_CHARS
+        return (self.apikey[:visible_chars] +
+                'X' * max(len(self.apikey) - visible_chars, 0))
+
+
+class Target(_Target):
 
+    def __new__(cls, project_id, endpoint, apikey, *args, **kwargs):
+        cls._inst = super(Target, cls).__new__(cls, project_id, endpoint,
+                                               APIkey(apikey), *args, **kwargs)
+        return cls._inst
 
```

Comment:
The tests fail because flake8 expects to see 2 lines b/w a class definition and a module constant, could you add it pls?  https://travis-ci.org/scrapinghub/shub/jobs/586109487

Reply:
Missed that. Pycharm didn't give any alert. Already fixed